### ISSUE TITLE
Unable to locate element by using its accessibility id/automation id in WPF desktop application using WinAppDriver

### ISSUE THREAD

**Issue by aakilb** [microsoft/WinAppDriver#1595]
Hello,

I am trying to automate a WPF desktop application using WinAppDriver but I am getting error messages and really need your help in this regard. I am working in Python and with the Gauge Test Automation framework.

I am able to start the application by automation. However, when I try to automate the clicking of a button after the application has started, I am getting the following error:

`Error Message: Message: An element could not be located on the page using the given search parameters.`

I am using the button's AutomationId i.e. AccessibilityId. I am using inspect.exe tool to identify the button's AutomationId which in this case is "NewButton". Thus, I am 100% certain that the AutomationId of this button is indeed "NewButton".

I am using the `find_element_by_accessibility_id()` function.

I request your kind help in this regard.

Thank you in advance.

**Comment by anunay1** [comment]
Please share the code and what is shown in inspect.

**Comment by aakilb** [comment]
Please find attached the code and the screenshot.
[code.txt](https://github.com/microsoft/WinAppDriver/files/7171481/code.txt)

![Screenshot](https://user-images.githubusercontent.com/77597121/133465491-d0e244f4-30bc-4c42-8b15-ab5d4df84b2c.png)

**Comment by anunay1** [comment]
Can you check how msny button with this automation Id exists

**Comment by aakilb** [comment]
How can I check?

**Comment by anunay1** [comment]
find_elements_by_accessibility_id()

**Comment by aakilb** [comment]
Yes. I had done that and I had found that the list is empty i.e. the length of the list is 0. Thus to answer your question, no buttons are located with this automation ID.

**Comment by aakilb** [comment]
I also tried automating clicking of "Maximize", but the same problem is happening. I believe, according to all my observations, that WinAppDriver is unable to detect any UI element in my application. 

Please advise.

Thank you.

**Comment by SystemCheater** [comment]
I had similar problems when WinAppDriver is not running as Administrator (or with apropriate privileges).

**Comment by aakilb** [comment]
@SystemCheater

I tried that but it did not solve the problem. Nevertheless, Thank you for your suggestion.

**Comment by aakilb** [comment]
@anunay1 

I really need your help. I have looked everywhere but I still can't figure this out. I am completely stuck. Please help me.

Thanks in advance.

**Comment by anunay1** [comment]
setup a zoom call anunayathakur1@gmail.com

**Comment by anunay1** [comment]
I am only available for next half and hour, consider sending the invitation fast.

**Comment by aakilb** [comment]
Hello,

I can't meet with you right now. I would like to meet with you sometime in the future and will let you know when I am ready.

Thanks.

**Comment by aakilb** [comment]
For anyone reading this in right now or in the future, I was able to find the solution of the problem.  You just need to put in some 2 to 3 seconds of wait time to give the application you're automating some time to start up. You can do this by passing in one more desired capability while opening the application and that is the 'ms:waitForAppLaunch'. Please refer to what is shown below.

`desired_caps["ms:waitForAppLaunch"] = 3`

The above code will wait for 3 seconds after launching the application and then try to locate the UI elements you're looking for.

Reference: https://github.com/appium/appium-windows-driver#:~:text=value%20is%2020000.-,ms%3AwaitForAppLaunch,-Similar%20to%20createSessionTimeout

**Comment by Tree55Topz** [comment]
@aakilb make sure to close the issue then.

**Comment by danielemarinelli** [comment]
Hello, (before closing this issue)
I am new with webappdriver, I receive NoSuchElement when trying to locate the LOCK button (see screen)
my code is with sleep of 10secs:
System.out.println("SignalsAdmin opens correctly and pressing Lock button...");
        Thread.sleep(10000);
        driverWinSA.findElementByName("Lock").click();
         //WebElement Lock = driverWinSA.findElementByAccessibilityId("1283");  //Lock button
        //Lock.click();

<img width="482" alt="lock" src="https://user-images.githubusercontent.com/64636628/134524302-024e5f5a-054e-41db-93fe-92579c9f5524.PNG">

what can the reason be? thanks

**Comment by Tree55Topz** [comment]
@danielemarinelli what is findElementByName? Is this a wrapper function you created? I thought the python version was find_element_by_name()

**Comment by danielemarinelli** [comment]
I wrote in Java my script, sorry if I made some confusion,  should I change topic?

**Comment by aakilb** [comment]
@danielemarinelli 

Hi.

The thread.sleep did not work for me (I did Python time.sleep()). What you need to do is to pass in the desired capability of "ms:waitForAppLaunch" of 3 seconds or similar.

**Comment by danielemarinelli** [comment]
ok thanks
